New Slint templates

tronical opened the pull request:
Merge the instructions from commontorizon/slint-{base|sdk} into the containers here and base the templates on torizon/wayland-base.

Use binary Slint packages for arm to speed up the C++ build.

cc #271

tronical committed:
WIP: new Slint templates

Merge the instructions from commontorizon/slint-{base|sdk} into the containers here
and base the templates on torizon/wayland-base.

Use binary Slint packages for arm to speed up the C++ build.

tronical committed:
Fix armhf build

tronical commented on rustSlint/Dockerfile.debug:
```diff
@@ -34,7 +33,8 @@ ARG GPU=
 ##
-FROM commontorizon/slint-base-${IMAGE_ARCH}${GPU}:${BASE_VERSION} AS debug
+FROM --platform=linux/${IMAGE_ARCH} \
+    torizon/wayland-base${GPU}:${BASE_VERSION} AS debug
```
Known issue here is that for this image to work, we also need to install libslint_cpp.so*. That was previously in the base image.

tronical committed:
Deply libslint_cpp.so also on the debug image

tronical committed:
Use Slint 1.9.2 instead of nightly

tronical commented:
Rebased onto current dev branch.

andreriesco commented on cppSlint/Dockerfile:
```diff
@@ -21,18 +20,86 @@ ARG APP_ROOT=
 
 # BUILD ------------------------------------------------------------------------
-FROM commontorizon/slint-sdk-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+FROM torizon/debian-cross-toolchain-${IMAGE_ARCH}${GPU}:${CROSS_SDK_BASE_TAG} AS build
```
The `${GPU}` can be removed here

tronical replied:
Thanks, that makes sense :)

andreriesco commented on cppSlint/Dockerfile.sdk:
```diff
@@ -19,12 +19,89 @@ ARG GPU=
+
 # BUILD ------------------------------------------------------------------------
-FROM commontorizon/slint-sdk-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+FROM torizon/debian-cross-toolchain-${IMAGE_ARCH}${GPU}:${CROSS_SDK_BASE_TAG} AS build
```
The `${GPU}` can be removed here

andreriesco commented on rustSlint/Dockerfile:
```diff
@@ -23,7 +23,7 @@ ARG GPU=
 
 # BUILD ------------------------------------------------------------------------
-FROM commontorizon/slint-sdk-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+FROM torizon/debian-cross-toolchain-${IMAGE_ARCH}${GPU}:${CROSS_SDK_BASE_TAG} AS build
```
The `${GPU}` can be removed here

andreriesco commented on rustSlint/Dockerfile.sdk:
```diff
@@ -21,7 +20,7 @@ ARG APP_ROOT=
 
 # BUILD ------------------------------------------------------------------------
-FROM commontorizon/slint-sdk-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+FROM torizon/debian-cross-toolchain-${IMAGE_ARCH}${GPU}:${CROSS_SDK_BASE_TAG} AS build
```
The `${GPU}` can be removed here

tronical committed:
Fix download links to Slint binary packages

microhobby commented:
Hey @tronical sorry for the delay, I was working to migrate the PowerShell scripts to Python/Xonsh. Now that this seems to be complete I will take a look here.

tronical commented:
No problem. I see that you merged the big PR to update all templates. Shall I rebase this PR before you start reviewing?

microhobby commented:
> Shall I rebase this PR before you start reviewing?
@tronical yes, please. I think that would be better than I rebase myself. We changed the major tags to `4` that is the images that support TOS 7.

tronical commented:
Ok, rebased on current dev and switched to version `4`. I build tested with `tasks.xsh run` manually through the different stages, but I'm a bit limited as the extension doesn't seem to build the resulting templates.

tronical commented:
Thanks for the review! Yay, now we have one container indirection less and it's easier to update slint version. Even users can do it easily now.
